src/plot.js

```diff
@@ -97,6 +97,18 @@ export function plot(options = {}) {
         ? facetsExclude || (facetsExclude = facetsIndex.map((f) => Uint32Array.from(difference(facetIndex, f))))
         : undefined;
     const {data, facets, channels} = mark.initialize(markFacets, facetChannels);
+    if (
+      facets && // if truthy, faceting is used
+      mark.facet === "auto" && // if the user didn't specify a facet
+      mark.data &&
+      (mark.data.length === facetIndex?.length || mark.data.size === facetIndex?.length) && // for Maps
+      mark.data !== facet?.data &&
+      facetIndex?.length > 1
+    ) {
+      warn(
+        `Facet data must strictly equal ${mark.ariaLabel} mark data for the appropriate faceting. This can be cause by using a filter function in the facet. If you intend for this mark to be faceted, set the facet option to true; otherwise set the facet option to false to disable this warning`
```

Comment:
```suggestion
        `Warning: the ${mark.ariaLabel} mark appears to use faceted data, but isn’t faceted. The mark data has the same length as the facet data and the mark facet option is "auto", but the mark data and facet data are distinct. If this mark should be faceted, set the mark facet option to true; otherwise, suppress this warning by setting the mark facet option to false.`
```